src/libstd/primitive_docs.rs

```diff
@@ -769,19 +769,58 @@ mod prim_str {}
 mod prim_tuple {}
 
 #[doc(primitive = "f32")]
-/// The 32-bit floating point type.
+/// A 32-bit floating point type (specifically, the "binary32" type defined in IEEE 754-2008).
+///
+/// This type can represent a wide range of decimal numbers, like `3.5`, `27`,
+/// `-113.75`, `0.0078125`, `34359738368`, `0`, `-1`. So unlike integer types
+/// (such as `i32`), floating point types can represent non-integer numbers,
+/// too.
+///
+/// However, being able to represent this wide range of numbers comes at the
+/// cost of precision: floats can only represent some of the real numbers and
+/// calculation with floats round to a nearby representable number. For example,
+/// `5.0` and `1.0` can be exactly represented as `f32`, but `1.0 / 5.0` results
+/// in `0.20000000298023223876953125` since `0.2` cannot be exactly represented
+/// as `f32`. Note however, that printing floats with `println` and friends will
+/// often discard insignificant digits: `println!("{}", 1.0f32 / 5.0f32)` will
+/// print `0.2`.
+///
+/// The precision is better for numbers near 0 and worse for large numbers. For
+/// example, above 2<sup>24</sup>, not even all integers are representable.
```

Comment:
In the IEEE 754 standard (which is very relevant in this context) precision has a very particular definition:

> **2.1.41 precision:** The maximum number *p* of significant digits that can be represented in a format, or the
number of digits to that a result is rounded.

With that definition, the precision is a property of the format, not of the number; and it would always be a precision of 24 bits for `f32`. Maybe a better word would be resolution, which I think means basically the same thing but is not explicitly overloaded by the IEEE 754 standard.

Reply:
That's a very good point. I am not sure if "resolution" would be better, however. But maybe that's just non-native me not properly understanding that word. Alternatively, what do you think about just removing this paragraph?

Reply:
Maybe something along the line of: Each float value corresponds/covers/maps to a range/interval of reals. Near 0 those ranges are smaller.

Reply:
I wasn't sure about it myself :) maybe it's only used in some domains and would not be clear in general documentation such as this. I'm thinking removing the paragraph is better.

Reply:
@the8472 What concerns me is that such statements are correct but would probably be understood by people already knowledgeable of floating point, while confusing those who aren't.

Reply:
The previous paragraph defines it as rounding to the nearest representable number. Then this paragraph could reference that definition and explain that while near zero the ranges of numbers rounded to each representable value are small they will grow as we move away from zero.

Reply:
I removed the paragraph now.